astropy/coordinates/baseframe.py

```diff
@@ -1370,34 +1371,21 @@ def _frameattr_equiv(left_fattr, right_fattr):
                     AstropyWarning,
                 )
                 return False
-            if isinstance(right_fattr, left_fattr.__class__):
-                # if same representation type, compare components.
-                return np.all(
-                    [
-                        (getattr(left_fattr, comp) == getattr(right_fattr, comp))
-                        for comp in left_fattr.components
-                    ]
-                )
-            else:
-                # convert to cartesian and see if they match
-                return np.all(
-                    left_fattr.to_cartesian().xyz == right_fattr.to_cartesian().xyz
+            return (
+                all(
+                    (getattr(left, comp) == getattr(right, comp)).all()
+                    for comp in left.components
                 )
-        elif left_is_repr or right_is_repr:
-            return False
-
-        left_is_coord = isinstance(left_fattr, BaseCoordinateFrame)
-        right_is_coord = isinstance(right_fattr, BaseCoordinateFrame)
-        if left_is_coord and right_is_coord:
-            # both are coordinates
-            if left_fattr.is_equivalent_frame(right_fattr):
-                return np.all(left_fattr == right_fattr)
-            else:
-                return False
-        elif left_is_coord or right_is_coord:
-            return False
+                if isinstance(right, type(left))
+                else np.all(left.to_cartesian().xyz == right.to_cartesian().xyz)
+            )
 
-        return np.all(left_fattr == right_fattr)
+        left_is_coord = isinstance(left, BaseCoordinateFrame)
+        return (
+            (left_is_coord is isinstance(right, BaseCoordinateFrame))
+            and (not left_is_coord or left.is_equivalent_frame(right))
+            and np.all(left == right)
+        )
```

Comment:
@mhvk, does this part look good to you?

Reply:
Actually, that has a bit the same problem; the previous definition of both `left_is_coord` and `right_is_coord` kept symmetry; to me, for something like `left_is_coord is isinstance(right, BaseCoordinateFrame)` I really have to think what the heck is going on, and that is not necessary here. The previous code was clearer, to me at least.  

I also feel these long return statements just work around the "not too many `return`s" rule -- this is why I've been pushing back a bit against enforcing all these new rules. (Though it is really good to see what they mean, since in some cases it is really obvious they make old cruft better -- as was clear from quite a few of your previous PRs...)

p.s. I suggested a `match` statement, but looking into that it does not seem a good match, sadly.